USM.Test/AlgorithmTest.cs

```diff
@@ -0,0 +1,509 @@
+                StartTime = 1000,
+                EndTime = 1050,
+                Code = "A02"
+            };
+            Meeting CSE27Discussion2 = new Meeting()
+            {
+                MeetingType = MeetingType.Discussion,
+                Days = (Days.Tuesday | Days.Thursday),
+                StartTime = 1530,
+                EndTime = 1700,
+                Code = "B02"
+            };
+            Course CSE27 = new Course()
+            {
+                CourseAbbreviation = "CSE27",
+                CourseName = "Intro / Discrete Mathematics",
+                Units = 4,
+                Description = ""
+            };
+            Section CSE27Section1 = new Section()
+            {
+                Ticket = 271,
+                Course = CSE27
+            };
+            Section CSE27Section2 = new Section()
+            {
+                Ticket = 272,
+                Course = CSE27
+            };
+
+            // Adding different sections to the course object
+            //adding for cse 20
+            CSE20.Sections.Add(CSE20Section1);
+            CSE20Section1.Meetings.Add(CSE20Lecture);
+            //adding for cse 21
+            CSE21.Sections.Add(CSE21Section1);
+            CSE21Section1.Meetings.Add(CSE21Lecture);
+            //adding for cse 22
+            CSE22.Sections.Add(CSE22Section1);
+            CSE22Section1.Meetings.Add(CSE22Lecture);
+            //adding for cse 23
+            CSE23.Sections.Add(CSE23Section1);
+            CSE23Section1.Meetings.Add(CSE23Lecture);
+            //adding for cse 24
+            CSE24.Sections.Add(CSE24Section1);
+            CSE24Section1.Meetings.Add(CSE24Lecture);
+            //adding for cse 25
+            CSE25.Sections.Add(CSE25Section1);
+            CSE25Section1.Meetings.Add(CSE25Lecture);
+            //adding for cse 26
+            CSE26.Sections.Add(CSE26Section1);
+            CSE26Section1.Meetings.Add(CSE26Lecture);
+            CSE26Section1.Meetings.Add(CSE26Discussion1);
+            CSE26Section2.Meetings.Add(CSE26Lecture);
+            CSE26Section2.Meetings.Add(CSE26Discussion2);
+            CSE26.Sections.Add(CSE26Section2);
+            //adding for cse 27
+            CSE27.Sections.Add(CSE27Section1);
+            CSE27Section1.Meetings.Add(CSE27Lecture);
+            CSE27Section1.Meetings.Add(CSE27Discussion1);
+            CSE27Section2.Meetings.Add(CSE27Lecture);
+            CSE27Section2.Meetings.Add(CSE27Discussion2);
+            CSE27.Sections.Add(CSE27Section2);
+
+            Course[] courses = { CSE20, CSE21, CSE22, CSE23, CSE24, CSE25, CSE26, CSE27 };
+            return courses;
+        }
+
+        //desperate single test
+        [Fact]
+        public void TestSingleClass()
+        {
+            //use the course above
+            ScheduleRepo sch = new ScheduleRepo();
+            Course[] AllCourses = Classes();
+
+            // choose the courses to be tested
+            Course CSE20 = AllCourses[0];
```

Comment:
Avoid using magic indexes, if you want to change a course, it can have serious consequences that are really hard to catch.